doc/decisions/operations_library.md

```diff
@@ -0,0 +1,50 @@
+# Operations Library
+
+## Problem
+
+There are many functions that can be useful, but are not required to form the minimal API.
+Since we want to keep the API of `libelektra-core` minimal, we cannot add these functions.
+
+## Constraints
+
+- `libelektra-core` should have a minimal (but still useful) API.
+
+## Assumptions
+
+## Considered Alternatives
+
+## Decision
+
+Create a new `libelektra-operations` library.
+This library provides generally useful, but not minimal, functions.
+The implementations should still be as efficient as possible and will therefore access private API directly.
+
+`libelektra-operations` only depends on `libelektra-core`.
+
+Move basic, but not minimal, functions from `libelektra-core` to `libelektra-operations`:
+
+- `keyNeedSync`
+- `keyGetRef`
+- `ksDeepDup`
```

Comment:
This creates another implication: libelektra-kdb will depend on `libelektra-operations`

Reply:
In general, I don't see this as a problem.  `libelektra-kdb` is a pretty complex library, it may be quite hard to implement all of its features just based on the minimal `libelektra-core` API. However, if `ksDeepDup` would be the only reason for this dependency, we should probably add `ksDeepDup` to the core.